packages/compile-solidity/src/compilerSupplier/loadingStrategies/VersionRange.ts

```diff
@@ -9,39 +9,77 @@ import solcWrap from "solc/wrapper";
 import { Cache } from "../Cache";
 import { observeListeners } from "../observeListeners";
 import { NoVersionError, NoRequestError } from "../errors";
+import { CompilerSupplier } from "@truffle/compile-common";
+import type { Results } from "@truffle/compile-solidity/compilerSupplier/types";
+import * as defaults from "@truffle/compile-solidity/compilerSupplier/defaults";
+
+export namespace VersionRange {
+  export type Specification = {
+    constructor: {
+      options: {
+        events: any;
+        compilerRoots?: string[];
+        solcConfig: {
+          version?: string;
+        }
+      }
+    };
+    results: Results.Specification;
+    allowsLoadingSpecificVersion: true;
+    allowsListingVersions: true;
+  };
+}
 
-export class VersionRange {
+export class VersionRange implements CompilerSupplier.Strategy<
+  VersionRange.Specification
+> {
```

Comment:
As @haltman-at mentions in [his review of #4222](https://github.com/trufflesuite/truffle/pull/4222#pullrequestreview-728040585), we should remove the method `findNewestValidVersion()` and replace it with `semver.maxSatisfying()`.